### ISSUE TITLE
[FEATURE] - Advanced Character Deletion/Restoration System

### ISSUE THREAD

**Issue by Gilroes** [TrinityCore/TrinityCore#28050]
### Description

I’d like to make a suggestion to have an advanced character deletion/restoration system. Currently, you only have the option to unlink deleted characters from the account or to purge them from the database entirely, while these options are useful, if can be tiresome to have to manually go through the database to delete characters below a certain level whilst keeping others that are at a higher level.

I did some research and I could not find exactly how Blizzard did it back in Wrath of the Lich King, but we could go off of this example: https://web.archive.org/web/20191009060813/https://us.battle.net/support/en/article/15312

As the default options and allow the server administrator to customise the criteria.

### Expected behaviour

N/A

### Steps to reproduce the problem

No steps to reproduce.

### Branch

3.3.5

### TC rev. hash/commit

222a0ba

### Operating system

Windows 11 x64

### Custom changes

None

**Comment by Aokromes** [comment]
you can select the level of the characters to be automatically deleted already.

**Comment by Gilroes** [comment]
> you can select the level of the characters to be automatically deleted already.

Yes but you can only define one level to be deleted automatically, such as level 59 characters after x amount of days l, I meant defining more than one retention policy.

As shown in the screenshot attached: 
![Uploading 6C8972BB-627F-49D6-A0AD-42F9B45F2721.png…]()

**Comment by Gilroes** [comment]
Could you please mark this as a bounty? Thank you.
